pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorSubscriptionTest.java

```diff
@@ -704,6 +705,66 @@ public void testReplicatedSubscriptionWhenReplicatorProducerIsClosed() throws Ex
         Awaitility.await().untilAsserted(() -> assertNotNull(topic2.getSubscription(subscriptionName)));
     }
 
+    @Test
+    public void testReplicateSubBackLog() throws Exception {
+        String namespace = BrokerTestUtil.newUniqueName("pulsar/testReplicateSubBackLog");
+        String topic = "persistent://" + namespace + "/when-replicator-producer-is-closed";
+        String subName = "sub";
+
+        admin1.namespaces().createNamespace(namespace);
+        pulsar1.getConfiguration().setReplicatedSubscriptionsSnapshotFrequencyMillis(100);
+        @Cleanup
+        PulsarClient client1 = PulsarClient.builder().serviceUrl(url1.toString())
+                .statsInterval(0, TimeUnit.SECONDS)
+                .build();
+        @Cleanup
+        Consumer<byte[]> consumer = client1.newConsumer()
+                .topic(topic)
+                .subscriptionName(subName)
+                .ackTimeout(5, TimeUnit.SECONDS)
+                .subscriptionType(SubscriptionType.Shared)
+                .replicateSubscriptionState(true)
+                .subscribe();
+        @Cleanup
+        Producer<byte[]> producer = client1.newProducer()
+                .topic(topic)
+                .create();
```

Comment:
Since the replicated replication will publish markers in these scenarios:
- Publish a marker per second if there are new messages are sent
- Ack message that after a snapshot

We'd better make this test flow like this:
- send some messages
- sleep `2s`
- ack messages
- send some messages again
- ack messages

---

And could you also add another test:
- Enable the feature replication on topic level or namespace level
- Pub & Sub
- Disable the feature replication.
- Verify all things work as expected.

Reply:
Thanks for your suggestion. I have added the ack message case into the test.